public/pages/workflows/new_workflow/utils.ts

```diff
@@ -138,68 +145,136 @@ export function fetchEmptyUIConfig(): WorkflowConfig {
   };
 }
 
-export function fetchSemanticSearchMetadata(): UIState {
+export function fetchSemanticSearchMetadata(version: string): UIState {
+  if (!version) {
+    throw new Error(
+      'Version parameter is required for fetchSemanticSearchMetadata'
+    );
+  }
```

Comment:
is this necessary? Typically the fn signatures should trust that the inputs/parameters are validated & correct before executing, otherwise it lowers the confidence and integrity of the code. How about preventing to run this fn if there is no version passed?

Reply:
good point about the validation.Removed the validation at the individual function level and handling to the caller. In this case, each individual function will have version as the parameter.

Reply:
addressed on line 152